### ISSUE TITLE
Feature-Request: Add a way to direct send BC from private keys

### ISSUE THREAD

**Issue by mowgli** [spesmilo/electrum#7244]
When using standard wallet, it is only possible to sweep private keys, not import them. But sweeping costs BC.

When getting bitcoins via paper wallet, it would be nice to direct use the private key to send BC (maybe with BC from addresses in standard wallet). But it is not possible to import private keys in such a type of wallet.

This would have several advantages:
* The rest could go to a change address
* One saves the additional cost
* Combined pays with already in store BC

The address/private key must be stored just until it is empty.

**Comment by SomberNight** [comment]
Several years ago, before version 2.0, Electrum supported this (importing individual privkeys to a seeded wallet). It was removed as too many newbies shot themselves in the foot. People did not understand what is backed up by the seed and what isn't.

-----

Still, I think maybe we could allow mixing HD and non-HD keys in an expert-only "advanced wallet" type that does not have a seed at all (see https://github.com/spesmilo/electrum/issues/6016).

Alternatively, we could allow "temporarily importing" privkeys/addresses. These would not get persisted in the wallet file, they would live in memory, and disappear once the process is closed. Using the Coins tab, the user could create all kinds of txs they want (e.g. combining the temporary UTXOs with wallet UTXOs, and sending change to the wallet).

I would prefer the seedless "advanced wallet" option though, as it is more generic and I think much of the work needed to do it would be useful for e.g. more advanced script type support.

**Comment by mowgli** [comment]
Hi,

Am So den 25. Apr 2021 um 15:46 schrieb ghost43:
> Several years ago, before version 2.0, Electrum supported this (importing individual privkeys to a seeded wallet). It was removed as too many newbies shot themselves in the foot. People did not understand what is backed up by the seed and what isn't.

Well, if it is hidden behind a expert option, it would be fine. There
could also be a big warning and the key could marked red in key list.

I don't think that a wallet should protect the user from his own
stupidiness. ;-)

> Still, I think maybe we could allow mixing HD and non-HD keys in an expert-only "advanced wallet" type that does not have a seed at all (see https://github.com/spesmilo/electrum/issues/6016).

Far beyond my idea. But as I read it, this would allow my use case too.

> Alternatively, we could allow "temporarily importing" privkeys/addresses. These would not get persisted in the wallet file, they would live in memory, and disappear once the process is closed. Using the Coins tab, the user could create all kinds of txs they want (e.g. combining the temporary UTXOs with wallet UTXOs, and sending change to the wallet).

This has some point too. However, the first option above might be
better as it is even a bit of safety.

Regards
   Klaus
-- 
Klaus Ethgen                                       http://www.ethgen.ch/
pub  4096R/4E20AF1C 2011-05-16            Klaus Ethgen <Klaus@Ethgen.ch>
Fingerprint: 85D4 CA42 952C 949B 1753  62B3 79D0 B06F 4E20 AF1C

**Comment by jonjonez** [comment]
I agree with the OP, we shouldn't protect the user from their own stupidity, a user prone to error will make mistakes in one way or another, limiting existing options or removing features because of stupidity harms those who already know how to use the tool.

Electrum is a wallet for advanced/experienced users.
